Source/WebCore/Modules/beacon/NavigatorBeacon.h

```diff
@@ -59,7 +59,7 @@ class NavigatorBeacon final : public Supplement<Navigator>, private CachedRawRes
     void notifyFinished(CachedResource&, const NetworkLoadMetrics&, LoadWillContinueInAnotherProcess) final;
     void logError(const ResourceError&);
 
-    Navigator& m_navigator;
+    WeakRef<Navigator> m_navigator;
```

Comment:
Is there something more specific that could be used for these Supplementable / Supplement cases? Its uniquely owned by the Navigator, so it seems like some less expensive constraint would be ideal.

Reply:
I mean, I could use `CheckedRef<>` too. Would you prefer that?